### PR TITLE
Restore backward compatibility of OSR command with maxit and tolf opt…

### PR DESCRIPTION
**Pull request by JohannesPfeifer**
…ions

Translates these options into optim_opt structure

### CODE REVIEW

File: matlab/osr1.m [hunk 1]
```diff
@@ -62,6 +62,27 @@
     error ('OSR: At least one of the initial parameter values for osr_params is NaN') ;
 end
 
+%restore backward compatibility with maxit and tolf
+if isfield(options_.osr,'maxit') || isfield(options_.osr,'tolf')
+    fprintf('\nOSR: The use of maxit and tolf is now deprecated. Please use the optim-option instead.\n')
```
Comment (houtanb) [inline matlab/osr1.m]: Can this be changed to a `warning`?

**Comment by houtanb** [comment]
@JohannesPfeifer I had understood that these options would no longer be included in the manual. Otherwise, for consistency, we would need to keep forever all deprecated options/commands in the manual.

**Comment by JohannesPfeifer** [comment]
@houtanb Ok, I will remove the manual entry. But searching for `deprecated` in the manual looks as if we keep all options around.

**Comment by JohannesPfeifer** [comment]
@houtanb I have done both changes

**Comment by MichelJuillard** [comment]
The usual procedure is first to declare an option or an instruction as deprecated in the manual and indicate its replacement. But the option or instruction is still working.
Then, a few versions later, we remove the option or the instruction altogether. Version 5.0 will be a good occasion to remove features declared obsolete a long time ago and introduce explicit error messages.
The compatibility code that we are maintaining should support old features forever.

**Comment by houtanb** [comment]
@MichelJuillard I totally agree with the process. My concern is that we never get to step two wherein we actually remove the entry, leaving things forever deprecated.

**Comment by houtanb** [comment]
@JohannesPfeifer sorry for the extra work, but please add back the manual entry as per @MichelJuillard 's comment

**Comment by JohannesPfeifer** [comment]
Ok. Done.

**Pull request merged**